pennylane/ops/op_math/controlled_ops.py

```diff
@@ -144,6 +144,17 @@ def _controlled(self, wire):
             work_wires=self.work_wires,
         )
 
+    @property
+    def has_decomposition(self):
+        if not super().has_decomposition:
+            return False
+        with qml.QueuingManager.stop_recording():
+            try:
+                self.decomposition()
+            except qml.operation.DecompositionUndefinedError:
+                return False
+        return True
+
```

Comment:
```suggestion
    @property
    def has_decomposition(self):
        if not all(self.control_values):
            return True
        if self.base.has_decomposition:
            return True
        return False
```

No more try-except.  The problem was with checking `self.base._controlled`.

Reply:
I chose my approach because I'd rather a try-except over copy-pasting code when we know it's not really a performance concern. [This](https://github.com/PennyLaneAI/pennylane/blob/1d3ce14d2a05c03225856045da176142bd6b2f41/pennylane/ops/op_math/controlled.py#L485) is how the current master evaluates it, but wrongly returns `True` because its base (`QubitUnitary`) has a `_controlled` method defined. Evidently, this is an insufficient indicator. Maybe we can remove that instead?

Reply:
what we really want is `hasattr(op.base, "_controlled") and op.base._controlled.return_type is not type(self)`, but of course, there is no `_controlled.return_type`

Reply:
Basically what is happening inside `Controlled.decomposition`:
```
    if len(op.control_wires) == 1 and hasattr(op.base, "_controlled"):
        result = op.base._controlled(op.control_wires[0])
        # disallow decomposing to itself
        # pylint: disable=unidiomatic-typecheck
        if type(result) != type(op):
            return [result]
```

Reply:
chatted offline - I'm worried that re-implementing here will make things fall out of sync (like they did before, which brings us here), so I've added a comment to explain that we're effectively doing that ^ but it can't be done without actually trying to decompose the op.